### PR TITLE
fix: surface source description in MCP tool description (#267)

### LINKED ISSUE #267: Add `description` field to [[sources]] in dbhub.toml for connection context
**Is your feature request related to a problem? Please describe.**

When managing multiple sources in `dbhub.toml`, it can be difficult for users and AI agents to understand what each connection is used for or what databases it contains. This can lead to confusion and mistakes, especially when several connections are similar.

**Describe the solution you'd like**

Add a `description` field to each `[[sources]]` entry in `dbhub.toml`, allowing users to provide a human-readable context for that connection (e.g., which databases are included, purpose, or special notes for agents).

**Example**

```toml
[[sources]]
id= "analytics-prod"
dsn = "postgres://user:pass@host:5432/analytics"
description = "Contains analytics-related databases (events, metrics, reporting). Used by BI dashboards and read-only analysis agents."
```

**Describe alternatives you've considered**

- Relying on naming conventions, but this is insufficient for clarity.
- Using external documentation, but this disconnects config from usage.

**Additional context**
- The `description` field should be optional, plain-text, and visible anywhere source information is displayed.

**Label**: enhancement

### PR DESCRIPTION
**Pull request by Isla-Liu**
## Summary

Makes `[[sources]].description` from `dbhub.toml` actually reach the MCP tool description that AI clients read. Previously, the field was parsed and surfaced via the REST `/api/sources` endpoint but was **completely ignored** by the MCP tool registration path, defeating its documented purpose: *"Helps AI models understand the purpose and contents of each database when multiple sources are configured."*

## Problem (Fixes #267)

When a user configures multiple sources in `dbhub.toml` with descriptions:

```toml
[[sources]]
id = "analytics_prod"
description = "Production read replica for BI / analytics queries"
dsn = "postgres://..."

[[sources]]
id = "payments"
description = "Payments ledger — PII, write-restricted"
dsn = "postgres://..."
```

The MCP tool descriptions that AI clients see are:

- `execute_sql_analytics_prod`: **"Execute SQL queries on the 'analytics_prod' postgres database"**
- `execute_sql_payments`: **"Execute SQL queries on the 'payments' postgres database"**

The user's carefully-written descriptions never reach the AI. Models must guess routing from the `id` alone.

This is exactly what issue #267 reported — and another user (@Diluka) confirmed in that thread: *"I asked the AI, and it said that the description wasn't in the tool's prompt"* — before the issue was closed as *"supported, doc was missing"*.

## Root cause

`SourceConfig.description` is typed in `src/types/config.ts` (L51) and parsed correctly, but:

- `src/utils/tool-metadata.ts` → `getExecuteSqlMetadata()` and `getSearchObjectsMetadata()` hardcode the template string and **never reference `sourceConfig.description`**.
- `src/tools/index.ts` → `registerExecuteSqlTool` / `registerSearchObjectsTool` pass the hardcoded template straight to `server.registerTool({ description, ... })`.
- `grep sourceConfig.description` across the repo returns **one** hit: `src/api/sources.ts` (the REST endpoint for the built-in web UI). Nothing in the MCP path.

PR #232 introduced the `description` field and its commit message says it's *"exposed via `/api/sources` for MCP clients"* — but `/api/sources` is an HTTP REST endpoint, not the MCP protocol. That's the scope gap this PR closes.

## Fix

Prepend `sourceConfig.description` (when set) to the generated description in both `getExecuteSqlMetadata()` and `getSearchObjectsMetadata()`. All existing technical signals — `[READ-ONLY MODE]`, `(limited to N rows)`, source id, `dbType` — are preserved unchanged.

**Before:**
```
Execute SQL queries on the 'analytics_prod' postgres database [READ-ONLY MODE] (limited to 1000 rows)
```

**After (with `description = \"Production read replica for BI / analytics queries\"`):**
```
Production read replica for BI / analytics queries. Execute SQL queries on the 'analytics_prod' postgres database [READ-ONLY MODE] (limited to 1000 rows)
```

Business context comes first (stronger routing signal for the AI), technical/safety context is preserved and unchanged.

### What's not touched

- **Custom tool descriptions** (`CustomToolConfig.description`): already work correctly, path unchanged. Custom tools carry their own required description; prepending the source description to them would override explicit user intent.
- **REST `/api/sources` endpoint**: `tool.description` reflects the new concatenated string, `source.description` still returns the raw user-provided value. Existing web UI behavior is preserved.
- **`title`, `annotations`**: unchanged.

## Test plan

- [x] New integration test: `src/utils/__tests__/tool-metadata.integration.test.ts` (4 tests) covering `getExecuteSqlMetadata` and `getSearchObjectsMetadata` for sources with and without description. Verifies the prefix appears first, and the original template + readonly note are retained.
- [x] Existing tests unchanged — all 22 tests in `src/api/__tests__/sources.integration.test.ts` still pass, including `toContain(source.id)` / `toContain(source.type)` assertions that now also see the prefix.
- [x] TypeScript compile: `pnpm build:backend` passes.
- [x] Full test suite: no new failures introduced. The 3 unrelated pre-existing Windows / environment-specific failures (`sqlite.integration.test.ts` for file-based readonly, `manager.test.ts` for `.ssh/config` path separator, `json-rpc-integration.test.ts` for `spawn pnpm ENOENT`) exist on `main` without this patch and are out of scope.

## Related

- Closes #267
- Scope gap originates in #232 (feat: add description field for TOML sources). This PR finishes what #232 started by extending the description propagation from the REST API to the MCP protocol.
- Orthogonal to #277 (reducing multi-source context via a `list_sources` meta tool). That design change would independently benefit from accurate per-source descriptions once this PR lands.

### CODE REVIEW

File: src/utils/tool-metadata.ts [hunk 1]
```diff
@@ -106,12 +106,15 @@ export function getExecuteSqlMetadata(sourceId: string): ToolMetadata {
     ? `Execute SQL (${dbType})`
     : `Execute SQL on ${sourceId} (${dbType})`;
 
-  // Determine description with more context
+  // Determine description with more context.
+  // Prepend the user-provided `description` from the source config (if set)
+  // so AI clients reading the MCP tool list see the source's purpose first.
+  const userDescPrefix = sourceConfig.description ? `${sourceConfig.description}. ` : "";
```
[the remark this answers is not in the document]
Comment (Isla-Liu) [reply src/utils/tool-metadata.ts]: Addressed in 055f2a5. Extracted the logic into a pure helper `buildSourceDescriptionPrefix(description)` that: (1) trims surrounding whitespace, (2) returns `""` for undefined/empty/whitespace-only inputs, and (3) appends only a space (no extra period) when the trimmed description already ends with `.`, `!`, or `?`. The same helper is now used in both `getExecuteSqlMetadata` and `getSearchObjectsMetadata` so the two sites can't drift. Added 10 pure unit tests in `src/utils/__tests__/tool-metadata.test.ts` covering each edge case (including the `"Prod DB."` double-period regression you flagged).

File: src/utils/tool-metadata.ts [hunk 2]
```diff
@@ -149,9 +152,12 @@ export function getSearchObjectsMetadata(sourceId: string): { name: string; desc
   const title = isSingleSource
     ? `Search Database Objects (${dbType})`
     : `Search Database Objects on ${sourceId} (${dbType})`;
+  // Prepend the user-provided `description` from the source config (if set)
+  // so AI clients reading the MCP tool list see the source's purpose first.
+  const userDescPrefix = sourceConfig.description ? `${sourceConfig.description}. ` : "";
```
[the remark this answers is not in the document]
Comment (Isla-Liu) [reply src/utils/tool-metadata.ts]: Same fix as above — both sites now call the shared `buildSourceDescriptionPrefix` helper in 055f2a5, so `getSearchObjectsMetadata` gets the same trim + punctuation-aware behavior for free.

File: src/utils/tool-metadata.ts [hunk 3]
```diff
@@ -39,6 +39,27 @@ export interface ToolMetadata {
...
+ *   "Prod DB"            -> "Prod DB. "
+ *   "Prod DB."           -> "Prod DB. "      (no double period)
+ *   "  Prod DB!  "       -> "Prod DB! "      (trimmed, no added period)
+ *   "Query me?"          -> "Query me? "
+ */
+export function buildSourceDescriptionPrefix(description: string | undefined): string {
+  const trimmed = description?.trim() ?? "";
+  if (!trimmed) return "";
+  return /[.!?]$/.test(trimmed) ? `${trimmed} ` : `${trimmed}. `;
+}
```
[the remark this answers is not in the document]
Comment (Isla-Liu) [reply src/utils/tool-metadata.ts:67]: Addressed in 8d63191. Agreed — `":."` is an artifact with no redeeming semantic reason. Extended the terminator regex to `/[.!?:]$/` so a trailing colon is treated as already-introducing what follows, yielding `"Details below: Execute SQL..."` instead. `;` and `)` remain non-terminators — they signal mid-sentence continuation (joining clauses / closing a parenthetical within the same sentence) so `. ` still fires for those.

File: src/utils/__tests__/tool-metadata.test.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    // is the user's intent and must not be altered.
+    expect(buildSourceDescriptionPrefix("Line 1\nLine 2")).toBe("Line 1\nLine 2. ");
+    expect(buildSourceDescriptionPrefix("Line A, Line B")).toBe("Line A, Line B. ");
+  });
+
+  it("does not treat non-sentence-ending punctuation as terminators", () => {
+    // These are mid-sentence / structural punctuation; the helper should
+    // still append ". " to produce a complete sentence boundary.
+    expect(buildSourceDescriptionPrefix("Details below:")).toBe("Details below:. ");
+    expect(buildSourceDescriptionPrefix("(read-only)")).toBe("(read-only). ");
```
[the remark this answers is not in the document]
Comment (Isla-Liu) [reply src/utils/__tests__/tool-metadata.test.ts:57]: Test updated in 8d63191. The `"Details below:"` case now asserts `"Details below: "` (space-only) and lives in a dedicated test alongside the other sentence-ending cases. The remaining `/[.!?:]$/` non-matches (`)` and `;`) stay in the 'non-terminators' test with a comment explaining why `:` was moved out.

**Review by tianzhou — APPROVED** [review]
LGTM. Thanks for the contribution.
